### ISSUE TITLE
Support Internal app sharing

### ISSUE THREAD

**Issue by bassemawny** [Triple-T/gradle-play-publisher#563]
Google just announced a new Internal app sharing track for build distribution. Are there any plans to support this feature?
It would be great to be able to upload apks and aabs directly via the gradle task and get the public URL as the output in case of success.

Link to what's new blog post: https://android-developers.googleblog.com/2019/05/whats-new-in-play.html

**Comment by SUPERCILEX** [comment]
Absolutely! I think new tasks called `publishShareableBundle` and `publishShareableApk` that build your app by default but also accept a prebuilt artifact folder would be best.

PS: can you come up with better names? 😊

Docs for reference: https://developers.google.com/android-publisher/api-ref/internalappsharingartifacts

**Comment by bassemawny** [comment]
I think `publishShareableBundle` and `publishShareableApk` are descriptive enough 😄

**Comment by Sloy** [comment]
Another possibility could be `publishInternalSharingBundle` and `publishInternalSharingApk` for more similarity with the feature name. 
But personally the names you proposed are good enough 😊

**Comment by SUPERCILEX** [comment]
I was hesitating to use the term internal because there's an internal track. Thanks for the suggestions though! I'm still undecided, so we'll see.

**Comment by nielsz** [comment]
This feature would be great!
I'm not too thrilled about `publishShareableApk` though. When you publish an APK to Production Track, it is shared by definition (to the whole world). So seeing this taskname scares me. `publishInternalSharingApk` would make more sense to me.

**Comment by SUPERCILEX** [comment]
@nielsz I'm just not a fan of how long that name is. New idea: `publishPrivateApk`/`publishPrivateBundle`. And how would people feel about `uploadPrivateApk`/`uploadPrivateBundle`?

The problem with `publish[...]` is that it could conflict with existing variants people already have. For example, what if someone already has an `internal` variant? That will generate a `publishInternalReleaseApk` task. And if the build type is `sharing` for whatever reason, we'll get `publishInternalSharingApk`. Also, the internal sharing task itself will generate `publishInternalReleaseInternalSharingApk` with just the internal variant.

TL;DR: I think `uploadPrivateApk` which could turn into `uploadInternalReleasePrivateApk` might be as good as it will get.

**Comment by SUPERCILEX** [comment]
Ok, I went ahead with this format: `upload[Variant]Private[Apk/Bundle]`. Please give it a try with the [snapshot builds](https://github.com/Triple-T/gradle-play-publisher#snapshot-builds).

And if someone really hates the naming, please let me know. It can still change before the release.

**Comment by abhimaanmadhav** [comment]
"uploadDebugPrivateBundle"  task gives an error Task 'uploadDebugPrivateBundle' not found in root project 
and i cant find in gradle tab also? am i doing something wrong
<img width="288" alt="Screenshot 2019-07-10 at 3 02 53 PM" src="https://user-images.githubusercontent.com/7856480/60958449-020bf900-a324-11e9-938e-32c9a5471b6f.png">

**Comment by SUPERCILEX** [comment]
@abhimaanmadhav Ah, I didn't realize you could upload debug internal sharing artifacts. I've created #619 to track this, thanks for bringing it up.
